enhancements/update/versioning-cincinnati-api.md

```diff
@@ -0,0 +1,168 @@
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+As of today, we declare the schema version in Cincinnati Graph URI. We currently, do not embed the version information
+in json schema. The enhancement will embed the version information in the json schema. This has the benefit of being
+easy to implement and to pass around if we ship Cincinnati JSON over non-HTTP channels. The enhancement will also begin
+versioning Cincinnati payloads by media type, for example `application/vnd.redhat.cincinnati.v1+json`.
+The move to supplying requested version in media type field will require us to move towards a generic un-versioned Graph
+URI, with the move, we'll still support old URI for foreseeable future.
+
+## Motivation
+
+### Goals
+1. Embed version information in json schema.
+2. Implement media type versioning for clients requesting a specific schema version of Cincinnati
+3. Move to a generic un-versioned URI for Cincinnati
+4. Cincinnati continues to support older clients after adding support for new clients for foreseeable future.
+5. Add metrics to identify hits on URIs and versions.
+
+### Non-Goals
+1. This enhancement does not attempt to drop the `v1` URI.
+
+
+## Proposal
+
+### User Stories
+
+### Implementation
+
+#### Version information in json schema
+Embed the version information in json schema. The updated json schema should look like:
+```json
+{
+    "version": 1,
+    "nodes": [],
+    "edges": []
+}
+```
+
+#### Media Type Versioning
+Currently, the versioning is done by using the versioned URI and does not support media type versioning.
+This enhancement allows users to request a specific version of the Cincinnati specification
+[proactively][proactive-negotiation] with [the `Accept` header][accept].  
+For example, `Accept: application/vnd.redhat.cincinnati.v1+json` says "I only understand v1 Cincinnati graphs;
+please give me that format if possible".  
+In the event where the media type is not provided, or `application/json` is provided, Cincinnati should return a default
+response that is compatible with the oldest supported version so as not to break old clusters from upgrading.
+
+In a similar space, the Open Container Initiative Image Format Specification
+[uses versioned media types][image-spec-media-types] to identify versions of configuration and other formats.
+
+
+#### URI Change
+The current URI follows `{URL}/v1/graph`. The new un-versioned URI will follow `{URL}/graph`. The existing URI can be
+redirected to the new URI. This can be done through cincinnati. In the case where an old URI is being called, cincinnati
+will by default return the graph that is compatible with the oldest supported version.
```

Comment:
The oldest being v1 specifically or would a backwards compatible v1.1 be available at this endpoint too?

Reply:
we are gonna consider both the endpoints as one. Backward compatible v1.1 will be available at this endpoint as well. Currently, we will be supporting `v1` as that's the oldest graph we know. But going forward we can make decisions when to drop specific versions based on their usage.